notifier/notifier.go

```diff
@@ -117,7 +120,10 @@ type Manager struct {
 	stopRequested chan struct{}
 
 	alertmanagers map[string]*alertmanagerSet
-	logger        log.Logger
+	// configToAlertmanagers keeps track of alertmanager sets for each AlertmanagerConfig,
+	// helping to avoid dropping known alertmanagers and re-use them without waiting for SD updates when applying the config.
+	configToAlertmanagers map[string]*alertmanagerSet
```

Comment:
no need to store this, we can build it on the fly in ApplyConfig, I'll give it a try.

Reply:
done